Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -3529,21 +3529,29 @@ private async Task<PipExecutionStep> ExecutePipStep(RunnablePip runnablePip)
             }
         }
 
-        private static void FlagSharedOpaqueOutputs(IPipExecutionEnvironment environment, ExecutionResult executionResult)
+        private void FlagSharedOpaqueOutputs(IPipExecutionEnvironment environment, ProcessRunnablePip process)
         {
+            // Select all declared output files
+            IEnumerable<(AbsolutePath path, bool isKnownToBeUnderSharedOpaque)> paths = process.Process
+                .FileOutputs
+                .Select(f => (path: f.Path, isKnownToBeUnderSharedOpaque: false));

```

Comment:
>isKnownToBeUnderSharedOpaque: false [](start = 44, length = 35)

I think it's possible to declare a an output file inside an opaque output:
file: foo\1.txx
directory: foo
If I'm correct, I do not think we can blindly set 'under shared opaque' for all the file outputs

Reply:
`isKnownToBeUnderSharedOpaque` is only relevant when the value is `true`; when the value is `false`, the `MakeSharedOpaqueOutputIfNeeded` method will explicitly check every path to see if it is under a shared opaque directory. 

I'll remove this flag since it appears to be very confusing.

Reply:
removed `force` and the linq stuff